src/lib/Microsoft.Health.Events/Telemetry/Exceptions/InvalidEventHubException.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+
+namespace Microsoft.Health.Events.Telemetry.Exceptions
+{
+    public class InvalidEventHubException :
+        Exception,
```

Comment:
I remember us discussing a base class to capture some common fields (IsClientError, IsRetryable etc.) as a general base exception type (i.e., not specific to EventHub exceptions). Are you planning to send that as a subsequent change, totally fine if so, was just curious where we landed with that.

Reply:
I was deferring it but I've added a base exception `IomtException` now.

Reply:
Thanks for making the change, this is much cleaner IMO!